### DISCUSSION TITLE
Sorting albums alphabetically?

### DISCUSSION THREAD

**Discussion by helloiamoliver** [SubmarinerApp/Submariner#151] [Q&A]
Hi!

I might be being really stupid here, but is it possible to sort alphabetically when in the album view? I don't love the random order which it seems to be using at the moment!

Also, is it possible to set it to cache the entire server's library offline? This feature would be really useful for when I'm not on my LAN.

I'm probably being really dense here, but please point me in the right direction!

Cheers!

**Comment by NattyNarwhal** [comment]
Sorry for being late, I was on vacation when you posted this.

The album view is like [Subsonic's](http://demo.subsonic.org/login.view?user=guest1&password=guest). The [API endpoint](http://www.subsonic.org/pages/api.jsp#getAlbumList) does support that, through to be useful, pagination would have to be added.

Downloading the entire library at once is something that's been requested before (GH-147). The annoying thing would be traversing the entire library in a controlled manner. If your server supports it, a workaround is making a smart playlist containing the tracks you care about.

**Comment by antonsizikov** [comment]
Hi!
I also have some thoughts about album sorting. It would be really nice to show album year and to have on option to sort them in that way, just like navidrome does.

**Reply by NattyNarwhal** [reply]
One big annoyance is that the Subsonic API doesn't consider years a property of the album, but of the track (which is more an ID3 level mistake). It's possible to fix that in Submariner though (probably via a derived property, fetch first track's year; hope that isn't an expensive operation in Core Data...).

Hilariously, despite that, the API *does* let you get albums by a year range. Go figure...

**Reply by eclecticpassions** [reply]
I'm new to using Submariner with Navidrome... Is there anyway to view "all songs" or "all albums"? All I could see are: random, newest, highest, frequent and recent in album view. I want to see "all" for albums and under Go>"By Songs". Is it hidden somewhere or it's not implemented? Thanks.
